### ISSUE TITLE
Rendering of tables got broken if there are tabs inside their cells when file scope option is activated

### ISSUE THREAD

**Issue by N0rbert** [rstudio/bookdown#960]
Using latest stable packages and two below Rmd files

https://gist.github.com/N0rbert/aacb9f4397bdedf8302a9ed321508c3f

I got strange table rendering as shown below (PDF as example):

![](https://i.stack.imgur.com/VNyME.png)

In the above image pipes (`|`) are rendered in plain.

Note: installation of bookdown from github fixes the issue, but I want to inform you about the problem with latest stable 0.20.

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('bookdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/bookdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

```
> xfun::session_info('bookdown')
R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.7 LTS, RStudio 1.3.1093

Locale:
  LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=ru_RU.UTF-8       
  LC_COLLATE=en_US.UTF-8     LC_MONETARY=ru_RU.UTF-8    LC_MESSAGES=en_US.UTF-8   
  LC_PAPER=ru_RU.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
  LC_TELEPHONE=C             LC_MEASUREMENT=ru_RU.UTF-8 LC_IDENTIFICATION=C       

Package version:
  base64enc_0.1.3 bookdown_0.20   digest_0.6.25   evaluate_0.14   glue_1.4.2     
  graphics_3.4.4  grDevices_3.4.4 highr_0.8       htmltools_0.5.0 jsonlite_1.7.1 
  knitr_1.30      magrittr_1.5    markdown_1.1    methods_3.4.4   mime_0.9       
  rlang_0.4.7     rmarkdown_2.3   stats_3.4.4     stringi_1.5.3   stringr_1.4.0  
  tinytex_0.26    tools_3.4.4     utils_3.4.4     xfun_0.17       yaml_2.2.1     
> 
```

**Comment by cderv** [comment]
Thank you for the report! 

> Note: installation of bookdown from github fixes the issue, but I want to inform you about the problem with latest stable 0.20.

I can verify this indeed works now in dev version. 

From a `git bisect` I can say

* It works in dev version because we revert a change made in 0.20 : 285cba41c50df9dc49479bb6e4f33bf83558b18d
* Before that the change was introduced in 593668b31bc83038c0d8c63895059517033bc3af with the file scope thing

That means if you set `options(bookdown.render.file_scope = TRUE)` before rendering to activate the `--file-scope` in pandoc, you'll reproduce. 

I am not sure using tabulation inside table cells is popular but it is good to know that this issue exist. This is something related to pandoc I think not bookdown directly

**Comment by N0rbert** [comment]
@cderv thanks. 

> I am not sure using tabulation inside table cells is popular but it is good to know that this issue exists. This is something related to pandoc I think not bookdown directly

The *tab* symbols here were created by copying table from some office document to Rmd document. 
Then I have added `|` to it to transform table to Markdown table. In some cells I have changed some whitespaces (really spaces and tabs) to spaces, but not all as we can see. So this is how this issue arose.

**Comment by yihui** [comment]
Thanks for the report! I plan to release a new version of **bookdown** to CRAN late this week or early next week (after a new version of **rmarkdown** is on CRAN).

**Comment by cderv** [comment]
This seems to be fixed in **bookdown** 0.21
